packages/cli-plugin-deploy-pulumi/src/utils/env/withEnvVariant.ts

```diff
@@ -0,0 +1,17 @@
+import { createConfiguration } from "./configuration";
+
+export interface IWithEnvVariantParams {
+    variant: string | undefined;
+}
+
+export const withEnvVariant = (params: IWithEnvVariantParams) => {
+    return createConfiguration(() => {
+        const variant = (params.variant || "").trim();
+        if (!variant || variant.length < 3) {
```

Comment:
Does this mean that if user set `a` as variant name, that it would not be in the config?

Would this silently happen?

Thinking if we need the `length < 3` check here. 🤔